### PR TITLE
Adding sagemaker pytorch model zoo example code

### PR DESCRIPTION
**Pull request by dikhulla**
*Issue #, if available:*

*Description of changes:*

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [X] I have read the [CONTRIBUTING](https://github.com/aws/amazon-sagemaker-examples/blob/master/CONTRIBUTING.md) doc and adhered to the example notebook best practices
- [X] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-sagemaker-examples/blob/master/README.md)
- [X] I have tested my notebook(s) and ensured it runs end-to-end
- [X] I have linted my notebook(s) and code using `black-nb -l 100 {path}/{notebook-name}.ipynb`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: introduction_to_applying_machine_learning/sagemaker_pytorch_model_zoo/inference.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+
+def input_fn(input_data, content_type):
+    """
+    Args:
+        input_data: the request payload serialized in the content_type format
+        content_type: the request content_type
+    """
+    if content_type == "application/x-image":
+        decoded = Image.open(io.BytesIO(input_data))
```
[the remark this answers is not in the document]
Comment (dikhulla) [reply introduction_to_applying_machine_learning/sagemaker_pytorch_model_zoo/inference.py:106]: An exception is thrown if `content_type` is not supported. 

ValueError(f"Type [{content_type}] not supported.")

**Comment by dikhulla** [comment]
<p>done</p>

---

 <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3660/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by dikhulla** [comment]
<p>done</p>

---

 <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3660/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by dikhulla** [comment]
<p>done</p>

---

 <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3660/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by dikhulla** [comment]
<p>done</p>

---

 <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3660/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by dikhulla** [comment]
<p>done</p>

---

 <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3660/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by dikhulla** [comment]
<p>done</p>

---

 <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3660/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Review by atqy — APPROVED** [review]

**Pull request merged**